tests/nn/misc/test_checkpoint_activations.py

```diff
@@ -40,33 +47,57 @@ def forward(self, x):
 
 
 class TestComparisonToPyTorch(unittest.TestCase):
-    def _test_checkpoint_wrapper(self, device, log_memory_usage=False):
-        def get_loss_and_gnorm(model):
-            torch.manual_seed(1)
-            input = torch.rand(2, 16, 32).requires_grad_(True).to(device)
+    def _test_checkpoint_wrapper(self, device):
+        def get_loss_and_gnorm(model, input):
+            ret = {}
+            ret["mem_0"] = get_cuda_mem_allocated()
             model.zero_grad()
             loss = model(input).sum()
+            ret["mem_after_fwd"] = get_cuda_mem_allocated()
             loss.backward()
+            ret["mem_after_bwd"] = get_cuda_mem_allocated()
             gnorm = torch.norm(torch.stack([torch.norm(p.grad.detach()) for p in model.parameters()]))
-            return {"loss": loss, "gnorm": gnorm}
+            ret["loss"] = loss.item()
+            ret["gnorm"] = gnorm.item()
+            return ret
 
+        input = torch.rand(2, 16, 32).requires_grad_(True)
         model = Model().to(device)
-        no_cpt = get_loss_and_gnorm(model)
+        no_cpt = get_loss_and_gnorm(model, input.to(device))
 
         model = Model(use_pytorch_checkpoint=True).to(device)
-        pyt_cpt = get_loss_and_gnorm(model)
+        pyt_cpt = get_loss_and_gnorm(model, input.to(device))
+
+        model = Model(use_fairscale_checkpoint=True).to(device)
+        fairscale_cpt = get_loss_and_gnorm(model, input.to(device))
+
+        model = Model(use_fairscale_checkpoint=True, offload_to_cpu=True).to(device)
+        fairscale_cpt_offload = get_loss_and_gnorm(model, input.to(device))
+
+        # Check for correctness.
         torch.testing.assert_allclose(no_cpt["loss"], pyt_cpt["loss"])
         torch.testing.assert_allclose(no_cpt["gnorm"], pyt_cpt["gnorm"])
 
-        model = Model(use_fairseq_checkpoint=True).to(device)
-        fairseq_cpt = get_loss_and_gnorm(model)
-        torch.testing.assert_allclose(no_cpt["loss"], fairseq_cpt["loss"])
-        torch.testing.assert_allclose(no_cpt["gnorm"], fairseq_cpt["gnorm"])
-
-        model = Model(use_fairseq_checkpoint=True, offload_to_cpu=True).to(device)
-        fairseq_cpt_offload = get_loss_and_gnorm(model)
-        torch.testing.assert_allclose(no_cpt["loss"], fairseq_cpt_offload["loss"])
-        torch.testing.assert_allclose(no_cpt["gnorm"], fairseq_cpt_offload["gnorm"])
+        torch.testing.assert_allclose(no_cpt["loss"], fairscale_cpt["loss"])
+        torch.testing.assert_allclose(no_cpt["gnorm"], fairscale_cpt["gnorm"])
+
+        torch.testing.assert_allclose(no_cpt["loss"], fairscale_cpt_offload["loss"])
+        torch.testing.assert_allclose(no_cpt["gnorm"], fairscale_cpt_offload["gnorm"])
+
+        # Check for memory usage for cuda only.
+        if device == torch.device("cpu"):
```

Comment:
maybe a skip_test annotation to run this only when cuda is available?

Reply:
this test shouldn't be skipped. we are just skipping part of the assertions here.